### DISCUSSION TITLE
Github hosted renovate app and aws-machine-image datasource

### DISCUSSION THREAD

**Discussion by TehesFR** [renovatebot/renovate#14968] [Request Help] [answered]
Hello 👋 

I'm currently trying to use the [aws-machine-image](https://docs.renovatebot.com/modules/datasource/#aws-machine-image-datasource) datasource with the Github hosted renovate app, and I don't see how to pass AWS environment variables to the renovate app in the documentation.

I know the datasource is in experimental state, and currently working with renovate self-hosted.
I just would like to know if this feature is already supposed to work with the Github hosted renovate app ? 
If yes, I would love to know how to pass these variables ?
And if not, I will be interested to know if it's planned to integrate it in the near future ?

Thank you very much.

**Comment by rarkins** [comment]
@markussiebert wrote the code so I will need his help on this one. Does it currently assume that the machine running Renovate already has AWS credentials set? e.g. via IAM role etc?

**Reply by markussiebert** [reply]
Yes, the runtime needs AWS credentials (web identity, AccessKey in environmnet variables, ...) 

I don't know what you mean with Github hosted something like Actions you run in your pipelines, you can use the environment variable feature and provide your Access Key and Secret Acces Key ID.

![image](https://user-images.githubusercontent.com/16692560/161949232-25120b14-a7c6-4c9b-8e01-78d93348e0df.png)

Like I do here for NPM Token, but you do it with

```bash
AWS_ACCESS_KEY_ID=...
AWS_SECRET_ACCESS_KEY=...
AWS_DEFAULT_REGION=...
```

**Comment by TehesFR** [comment, marked as answer]
Hi @markussiebert thank you for your answer.

By "Github hosted renovate app" I mean that we've just enabled the renovate app, in the Github apps section here
![CleanShot 2022-04-06 at 13 45 34](https://user-images.githubusercontent.com/5724684/161969336-8bb671bf-fa19-4397-b42e-e3c9ee9fdb36.png)

By doing this, I don't know if we can provide environment variables to renovate, as we are not running the bot ourselves.

**Reply by rarkins** [reply]
No, you can't provide custom env to a hosted Renovate currently

**Reply by TehesFR** [reply]
OK thanks @rarkins
